File: 3254/Solution.java
class Solution {
    public int[] resultsArray(int[] nums, int k) {
        int n = nums.length;
        int[] a = new int[n];
        for(int i = 1; i < n; i++){
            a[i] = a[i-1];
            if(nums[i-1] != nums[i] - 1) a[i]++;
        }
        
        int[] res = new int[n - k + 1];

        for(int i = 0; i< res.length; i++){
            if(a[i] == a[i+k-1]){
                res[i] = nums[i+k-1];
            }else{
                res[i] = -1;
            }
        }
        return res;
    }
}

File: 3254/Solution.rs
impl Solution {
    pub fn results_array(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let n = nums.len();
        let k = k as usize;
        let mut a = vec![0; n];
        for i in 1..n {
            a[i] = a[i-1];
            if nums[i] - 1 != nums[i-1] {
                a[i] += 1;
            }
        }
        let mut res = vec![-1; n - k + 1];
        for i in 0..res.len() {
            if a[i] == a[i + k - 1] {
                res[i] = nums[i + k - 1];
            }
        }
        res
    }
}
